appinterface/projection/crossfire/crossfire.go

```diff
@@ -461,9 +485,115 @@ func (projection *Crossfire) checkTaskNetworkProposalVote(
+		if sender.Type == constants.TYPE_URL_PUBKEY && sender.MaybeThreshold == nil {
+
+			// Increment count for address as per PHASE
+			phaseAddressCountDbKey := phaseNumberPrefix + constants.DB_KEY_SEPARATOR + sender.Pubkeys[0]
+			errIncrementing := crossfireValidatorStatsView.IncrementOne(phaseAddressCountDbKey)
+			if errIncrementing != nil {
+				return fmt.Errorf("error Phase wise tx sent count increment: %v", errIncrementing)
+			}
+
+			// Increment TOTAL count for address
+			totalAddressCountDbKey := constants.TOTAL_TX_SENT_PREFIX + constants.DB_KEY_SEPARATOR + sender.Pubkeys[0]
+			errIncrementingTotal := crossfireValidatorStatsView.IncrementOne(totalAddressCountDbKey)
+			if errIncrementingTotal != nil {
+				return fmt.Errorf("error Incrementing tx sent count: %v", errIncrementingTotal)
+			}
+
 		}
 	}
+	return nil
+}
+
+// Compute Tx Sent Rank for participating validators
+func (projection *Crossfire) computeTxSentRank(
+	crossfireValidatorStatsView *view.CrossfireValidatorsStats,
+	crossfireValidatorView *view.CrossfireValidators,
+) error {
+	// Using dummy list for development
+	participantsList, errGetRequest := projection.Client.Participants()
+	if errGetRequest != nil {
+		return fmt.Errorf("[error] Getting Participants list %w", errGetRequest)
+	}
+
+	// Get All Tx Count Sorted
+	dbParticipantWithCountList, errDbCount := crossfireValidatorStatsView.FindAllLike(constants.TOTAL_TX_SENT_PREFIX)
+	if errDbCount != nil {
+		return fmt.Errorf("[error] Database Participant With Total Count %w", errDbCount)
+	}
+	if len(dbParticipantWithCountList) < 1 {
+		// No Participants to Rank
+		return nil
+	}
 
+	// Sort With Most Tx Sent First
+	sort.SliceStable(dbParticipantWithCountList, func(i, j int) bool {
+		return dbParticipantWithCountList[i].Value > dbParticipantWithCountList[j].Value
+	})
+
+	var rank int = 1
+	for index, dbParticipant := range dbParticipantWithCountList {
+		dbParticipantPrimaryAddress := strings.Split(dbParticipant.Key, constants.DB_KEY_SEPARATOR)[1]
+		// Check for each Participant in URL
+		for _, participant := range *participantsList {
+
+			// Check if the primary address match
+			if participant.PrimaryAddress == dbParticipantPrimaryAddress {
+
+				// Check if Participant is a Registered Crossfire Validator
+				registeredCrossfireValidator, errFindingValidator := crossfireValidatorView.FindBy(view.CrossfireValidatorIdentity{
+					MaybeOperatorAddress: &participant.OperatorAddress,
+				})
+				if errFindingValidator != nil || registeredCrossfireValidator == nil {
+					break
+				}
+
+				// Update Ranks with specific Validator
+				errUpdating := crossfireValidatorView.UpdateTxSentRank(rank, participant.PrimaryAddress, participant.OperatorAddress)
+				if errUpdating != nil {
+					return fmt.Errorf("[error] Updating TX SENT Task Rank %w", errUpdating)
+				}
+				if index+1 < len(dbParticipantWithCountList) && dbParticipantWithCountList[index].Value != dbParticipantWithCountList[index+1].Value {
```

Comment:
@cdc-Hitesh I'm a little bit confused about this logic, does it mean that `only current tx_count not equal to next tx_count` then `rank++`? If so, then the result will be `1, 2, 3, 4, 4, 4, 5` instead of `1, 2, 3, 4, 4, 4, 7` right?
I think we can keep the `rank++` for every loop and maintain a variable for `last_index_tx_count` and `last_rank`, so if the count are equal, use the `last_rank`, if not use the `rank`. Do you think this is feasible?

Reply:
@foreseaz Thanks.
>If so, then the result will be 1, 2, 3, 4, 4, 4, 5 instead of 1, 2, 3, 4, 4, 4, 7 right?

Yes, you are correct.
Can I understand a dataset where this can happen `1, 2, 3, 4, 4, 4, 7`? Because after rank `#_4` only #_5 is supposed to come right?